File: tests/adapters/cpp/goldens/function_bodies/max_tokens_trim.cpp
/**
 * C++ module for testing function body optimization.
 */

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

struct User {
    int id;
    std::string name;
    std::string email;
};

struct ProcessingResult {
    std::vector<User> valid;
    std::vector<User> invalid;
};

class Calculator {
private:
    std::vector<std::string> history;
    std::string name;

public:
    Calculator() : Calculator("default") {}

    Calculator(const std::string& name) : name(name) {
        history.reserve(10);
    }

    int add(int a, int b) {
        int result = a + b;
        // … method body truncated (2 lines)
        return result;
    }

    int multiply(int a, int b) {
        int result = a * b;
        // … method body truncated
        return result;
    }

    std::vector<std::string> getHistory() const {
        return history;
    }

private:
    bool validateInput(int value) {
        // … method body truncated (11 lines)

        return true;
    }
};

ProcessingResult processUserData(const std::vector<User>& users) {
    ProcessingResult result;

    // … function body truncated (9 lines)

    return result;
}

// Template function
template<typename T, typename Processor>
std::vector<T> processArray(const std::vector<T>& items, Processor processor) {
    std::vector<T> result;

    // … function body truncated (7 lines)

    return result;
}

// Lambda usage function
std::vector<int> filterPositive(const std::vector<int>& numbers) {
    std::vector<int> result;
    // … function body truncated (2 lines)
    return result;
}

int main() {
    Calculator calc("test");
    // … function body truncated (8 lines)

    return 0;
}

File: tests/adapters/rust/goldens/function_bodies/max_tokens_trim.rs
// Rust module for testing function body optimization.

use std::collections::HashMap;

#[derive(Debug, Clone)]
struct User {
    id: i32,
    name: String,
    email: String,
}

#[derive(Debug)]
struct ProcessingResult {
    valid: Vec<User>,
    invalid: Vec<User>,
}

struct Calculator {
    name: String,
    history: Vec<String>,
}

impl Calculator {
    fn new() -> Self {
        Self::with_name("default")
    }

    fn with_name(name: &str) -> Self {
        Self {
            name: name.to_string(),
        // … method body truncated (3 lines)
    }

    fn add(&mut self, a: i32, b: i32) -> i32 {
        let result = a + b;
        // … method body truncated (5 lines)
    }

    fn multiply(&mut self, a: i32, b: i32) -> i32 {
        let result = a * b;
        // … method body truncated (4 lines)
    }

    fn get_history(&self) -> Vec<String> {
        self.history.clone()
    }

    fn validate_input(&self, value: i32) -> bool {
        let value_str = value.to_string();

        // … method body truncated (14 lines)
    }
}

fn process_user_data(users: Vec<User>) -> ProcessingResult {
    let mut result = ProcessingResult {
        valid: Vec::new(),
    // … function body truncated (12 lines)
}

fn process_array<T, F>(items: Vec<T>, processor: F) -> Vec<T>
where
    F: Fn(T) -> T,
{
    let mut result = Vec::new();

    for item in items {
    // … function body truncated (5 lines)
}

fn filter_positive(numbers: Vec<i32>) -> Vec<i32> {
    numbers.into_iter().filter(|&n| n > 0).collect()
}

fn main() {
    let mut calc = Calculator::with_name("test");
    // … function body truncated (18 lines)
}
